src/Symfony/Component/Security/Core/Validator/Constraints/UserPasswordValidator.php

```diff
@@ -53,22 +52,13 @@ public function validate(mixed $password, Constraint $constraint)
 
         $user = $this->tokenStorage->getToken()->getUser();
 
-        if (!$user instanceof UserInterface) {
-            throw new ConstraintDefinitionException('The User object must implement the UserInterface interface.');
```

Comment:
is it intended to remove this error ?

Reply:
Yes, we only need `getPassword()` here so the check is replaced accordingly